kore/app/exec/Main.hs

```diff
@@ -620,59 +635,87 @@ mainDispatch = warnProductivity . mainDispatchWorker
                 then koreBmc localOptions proveOptions
                 else koreProve localOptions proveOptions
         | Just searchOptions <- koreSearchOptions =
-            koreSearch localOptions searchOptions
+            koreSearch exeLastModifiedTime localOptions searchOptions
+        | True <- serialize =
+            koreSerialize localOptions
         | otherwise =
-            koreRun localOptions
+            koreRun exeLastModifiedTime localOptions
       where
         KoreExecOptions{koreProveOptions} = execOptions
         KoreExecOptions{koreSearchOptions} = execOptions
+        KoreExecOptions{serialize} = execOptions
 
 koreSearch ::
+    UTCTime ->
     LocalOptions KoreExecOptions ->
     KoreSearchOptions ->
     Main (KFileLocations, ExitCode)
-koreSearch LocalOptions{execOptions, simplifierx} searchOptions = do
+koreSearch exeLastModifiedTime LocalOptions{execOptions, simplifierx} searchOptions = do
     let KoreExecOptions{definitionFileName} = execOptions
-    definition <- loadDefinitions [definitionFileName]
     let KoreExecOptions{mainModuleName} = execOptions
-    mainModule <- loadModule mainModuleName definition
+    let KoreExecOptions{koreSolverOptions} = execOptions
+    SerializedDefinition{serializedModule, lemmas, locations} <-
+        deserializeDefinition simplifierx koreSolverOptions definitionFileName mainModuleName exeLastModifiedTime
```

Comment:
If the definition wasn't serialized before, it will serialize it by default?

Reply:
I'm not entirely sure what you're asking. the `deserializeDefinition` function returns a `SerializedDefinition` object which contains everything needed to do execution or search with the haskell backend. What it does is detect whether the file it's reading is a serialized binary file or a textual kore definition and if it is the former, it simply deserializes it, and if it is the latter, it does all the work done by `kore-exec --serialize`, but does not actually write anything to disk, merely using the result to do execution.